### PR TITLE
Can get cached bitcoin balances and block info from CoreBitcoin (Issue #98)

### PR DESCRIPTION
**Pull request by willy-b**
To address #98 

Makes it so CoreBitcoin issues "balance" events even if balance is unchanged whenever it checks the balance, to support stateless/new consumers.

### CODE REVIEW

**Comment by willy-b** [comment]
tl;dr I'm going back but storing the last value in DattCore

Full plan:
The way I "fixed" this was stupid: You have to wait the polling interval after opening/closing the settings pane. For performance reasons the polling interval should be kept long, but for UX reasons, the refresh interval should be short. Thus these should be totally decoupled.
Better solution: I'm going to keep the balance state at the DattCore level, and add a method to get the current balance. The maximum UI wait is then the polling interval, experienced ONLY if they opened the settings immediately before the first poll. *Almost always* one will get the last polled value instantly from DattCore.

**Comment by willy-b** [comment]
Almost ready for review. 
Just pending a couple test additions and an update:
  CoreBitcoin#asyncGetLatestBlockInfo: (need to update to verify that 'block-info' event is emitted)
  DattCore/CoreBitcoin#getLastBlockInfo: (new test needed)
  DattCore#monitorCoreBitcoin: (need to update to verify 'bitcoin-block-info' event is emitted)

File: core/core-bitcoin.js [hunk 1]
```diff
@@ -138,8 +138,28 @@ CoreBitcoin.prototype.asyncPollBalance = function () {
...
 CoreBitcoin.prototype.asyncGetLatestBlockInfo = function () {
-  return this.blockchainAPI.asyncGetLatestBlockInfo()
+  return asink(function *() {
+    let blockInfo = yield this.blockchainAPI.asyncGetLatestBlockInfo()
+
+    let blockInfoFound = blockInfo && blockInfo.hashbuf
+    let blockChanged = blockInfoFound && (!this.lastBlockInfo || !this.lastBlockInfo.hashbuf || this.lastBlockInfo.hashbuf.toString('hex') !== blockInfo.hashbuf.toString('hex'))
+
+    if (blockChanged) {
+      this.emit('block-info', blockInfo)
```
Comment (ryanxcharles) [inline core/core-bitcoin.js]: This seems like poor practice to me to emit an event inside a "get" method. The only thing the "get" function should do is get the value. Perhaps the proper place for this code to be is inside the blockchainAPI code, which could be set to monitor for the latest block info. And in that case, when an update is found, an event is emitted.

File: react/box-bitcoin.jsx [hunk 2]
```diff
@@ -49,7 +49,16 @@ let BoxBitcoin = React.createClass({
 
   monitorDattCore: function () {
     let dattcore = this.props.dattcore
+    let initBalances = dattcore.getLastBalances()
+    if (initBalances) {
+      this.handleBitcoinBalance(initBalances)
```
Comment (ryanxcharles) [inline react/box-bitcoin.jsx]: The idea of the "monitor" function is to just have all the .on methods. There should probably be a different function for "emit initial events". Then the handle functions will handle them like normal. But then again, I'm not sure the UI should be emitting any initial events. The initial data should just be plugged into the props, no?

**Comment by willy-b** [comment]
Thanks for the review. Addressed your comments, let me know if you need anything else

**Comment by ryanxcharles** [comment]
Project has been rearchitected - PR no longer applies.
